.github/workflows/validate.yml

```diff
@@ -0,0 +1,24 @@
+name: ValidateTrialJsonSchema
+on:
+  push:
+    branches: [ feature/json-validation ]
+  pull_request:
+    branches: [ feature/json-validation ]
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: 'Log level'
+        required: true
+        default: 'warning'
+
+jobs:
+  Validate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Validate JSON
+        uses: Zingabopp/JsonValidate-Action@v1
+        with:
+          json-schema: sample_trial/trial-json-schema.json
```

Comment:
needs to be dynamic.
a) the schema needs to have its own folder. so all will reuse it
b) you need to calculate which files were changed and apply on those
b.2 ) option 2, you do that for each trial directory...

Reply:
a) Done
b) Done as well, please review and if accepted we can also change it to ignore the schema file in your workflows too.